service/history/transfer_queue_active_task_executor.go

```diff
@@ -838,6 +839,40 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 		}
 	}
 
+	// Note: childStarted flag above is computed from the parent's history. When this is TRUE it's guaranteed that the child was succesfully started.
+	// But if it's FALSE then the child *may or maynot* be started (ex: we failed to record ChildExecutionStarted event previously.)
+	// This is not an issue in the general flow since starting a child is an idempotent operation (guarded by childInfo.CreateRequestId)
+	// But in the case of resets, we overwrite childInfo.CreateRequestId with a newly generated ID. Hence we need special handling when we are
```

Comment:
I think this doesn't matter as the createRequestID check when starting a workflow only applies the latest run. 

To really solve the issue, StartWorkflow api needs to support start a specific runID and return an error if that run already exists.

Reply:
Done. Updated the comment.